jcl/src/jdk.management/share/classes/com/ibm/lang/management/ProcessorUsage.java

```diff
@@ -80,17 +80,32 @@ public ProcessorUsage() {
 	private ProcessorUsage(long user, long system, long idle, long wait,
 			       long busy, int id, int online, long timestamp) throws IllegalArgumentException {
 		super();
-		if ((user < -1) ||
-		   (system < -1) ||
-		   (idle < -1) ||
-		   (wait < -1) ||
-		   (busy < -1) ||
-		   (id < -1) ||
-		   (online < 0) || (online > 1) ||
-		   (timestamp < 0) ||
-		   ((user >= 0) && (system >= 0) && (wait >= 0) && (busy >= 0) && (busy < (user + system + wait))))
-		{
-			throw new IllegalArgumentException();
+		if (user < -1) {
```

Comment:
Does it make sense to also include the "bad" value in the exception message?

Reply:
Excellent idea